mono/mini/abcremoval.h

```diff
@@ -179,19 +179,28 @@ typedef struct MonoRelationsEvaluationRanges {
 
 /**
  * The context of a variable evaluation.
- * status: the evaluation status
  * current_relation: the relation that is currently evaluated.
  * ranges: the result of the evaluation.
  * father: the context of the evaluation that invoked this one (used to
  *         perform the backtracking when loops are detected.
  */
 typedef struct MonoRelationsEvaluationContext {
-	MonoRelationsEvaluationStatus status;
 	MonoSummarizedValueRelation *current_relation;
 	MonoRelationsEvaluationRanges ranges;
 	struct MonoRelationsEvaluationContext *father;
 } MonoRelationsEvaluationContext;
 
+/**
+ * This is a performance optimization. Clean_context was
+ * coming to dominate the running time of abcremoval. By
+ * storing the statuses together, we can memset the entire
+ * region.
+ */ 
+typedef struct MonoRelationsEvaluationContextContainer {
+	MonoRelationsEvaluationStatus *statuses;
+	MonoRelationsEvaluationContext *contexts;
+} MonoRelationsEvaluationContextContainer;
+
 /*
```

Comment:
Can we just store these fields in MonoVariableRelationsEvaluationArea, and pass that around ?